port_ocean/utils/queue_utils.py

```diff
@@ -0,0 +1,78 @@
+import asyncio
+from asyncio import Queue, Task
+from typing import Any, TypeVar, Callable, Coroutine
+
+from loguru import logger
+
+T = TypeVar("T")
+
+
+async def _start_processor_worker(
+    queue: Queue[Any | None],
+    func: Callable[..., Coroutine[Any, Any, T]],
+    results: list[T],
+) -> None:
+    while True:
+        raw_params = await queue.get()
+        try:
+            if raw_params is None:
+                return
+            logger.debug(f"Processing {raw_params[0]}")
+            results.append(await func(*raw_params))
+        finally:
+            queue.task_done()
+
+
+async def process_in_queue(
+    objects_to_process: list[Any],
+    func: Callable[..., Coroutine[Any, Any, T]],
+    *args: Any,
+    concurrency: int = 50,
+) -> list[T]:
+    """
+    This function takes a list of objects and process the within a queue buffer
+    allowing for smoother handling of bursts of incoming objects
+    This can help prevent overload and improve memory issues when dealing with large sset of data
```

Comment:
```suggestion
    This function executes multiple asynchronous tasks in a bounded way (e.g. having 200 tasks to execute, while running only 20 concurrently), to prevent overload and improve memory issues when dealing with large sets of data and tasks.
```